File: career-connect-job-portal/job-portal-frontend/src/services/jobService.js
import API from "../api/axios";

export const getAllJobs = () => {
    return API.get("/jobs");
};

export const getJobById = (id) => {
    return API.get(`/jobs/${id}`);
};

export const addJob = (job) => {
    return API.post("/jobs", job);
};

export const updateJob = (id, job) => {
    return API.put(`/jobs/${id}`, job);
};

export const deleteJob = (id) => {
    return API.delete(`/jobs/${id}`);
};

export const searchJobs = (keyword) => {
    return API.get(`/jobs/search?keyword=${keyword}`);
};

export const getJobsByLocation = (location) => {
    return API.get(`/jobs/location/${location}`);
};

export const getJobsByType = (type) => {
    return API.get(`/jobs/type/${type}`);
};

export const getJobsByExperience = (experience) => {
    return API.get(`/jobs/experience/${experience}`);
};

 

File: career-connect-job-portal/jobportal/src/main/java/com/onkar/jobportal/service/JobService.java
package com.onkar.jobportal.service;

import java.util.List;

import com.onkar.jobportal.dto.ApiResponse;
import com.onkar.jobportal.dto.JobRequest;
import com.onkar.jobportal.entity.Job;

public interface JobService {

    ApiResponse addJob(JobRequest request);

    List<Job> getAllJobs();

    Job getJobById(Long id);

    ApiResponse updateJob(Long id, JobRequest request);

    ApiResponse deleteJob(Long id);

    List<Job> searchJobs(String keyword);

    List<Job> getJobsByLocation(String location);

    List<Job> getJobsByJobType(String jobType);

    List<Job> getJobsByExperience(String experience);

}